cinema/urls.py

```diff
@@ -1,10 +1,44 @@
-from django.urls import path
+from django.urls import path, include
+from rest_framework import routers
 
-from cinema.views import movie_list, movie_detail
+from cinema.views import (
+    MovieViewSet,
+    GenreList,
+    GenreDetail,
+    ActorList,
+    ActorDetail,
+    CinemaHallViewSet,
+)
+
+router = routers.DefaultRouter()
+router.register("movies", MovieViewSet)
+
+cinema_hall_list = CinemaHallViewSet.as_view({"get": "list", "post": "create"})
+cinema_hall_detail = CinemaHallViewSet.as_view(
+    {
+        "get": "retrieve",
+        "put": "update",
+        "patch": "partial_update",
+        "delete": "destroy",
+    }
+)
 
 urlpatterns = [
-    path("movies/", movie_list, name="movie-list"),
-    path("movies/<int:pk>/", movie_detail, name="movie-detail"),
+    path("", include(router.urls)),
+    path("genres/", GenreList.as_view(), name="genre-list"),
+    path(
+        "genres/<int:pk>/",
+        GenreDetail.as_view(),
+        name="movie-detail",
+    ),
+    path("actors/", ActorList.as_view(), name="actor-list"),
+    path("actors/<int:pk>/", ActorDetail.as_view(), name="actor-detail"),
+    path("cinema_halls/", cinema_hall_list, name="cinema_hall-list"),
```

Comment:
also, maybe it would be better to use `_` or `-` only? not to mix them

Reply:
Yeah, sometimes unchecked find-and-replace technique doesn't work that well. :)